src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -1070,5 +1070,16 @@ protected bool IsFeatureEnabled(MessageID feature)
         {
             return this.Options.IsFeatureEnabled(feature);
         }
+
+        // TODO REVIEW
+        protected bool IsMakingProgress(ref int lastTokenOffset)
```

Comment:
Rather than returning false when we're not making progress, I'd rather this method throw an exception. That way it would help us find bugs rather than hiding bugs.